src/api/app/models/channel.rb

```diff
@@ -168,7 +168,10 @@ def add_channel_repos_to_project(tpkg, mode = nil)
       next unless mode == :enable_all || !ct.disabled
       # add repositories
       unless cp.project.repositories.find_by_name(repo_name)
-        tpkg.project.add_repository_with_targets(repo_name, ct.repository, [ct.repository])
+        unless tpkg.project.repositories.where(name: repo_name).exists?
```

Comment:
I guess you should use `.find_or_create_by` and then in all cases add the target. Not only in the case the `repositoriy` doesn't exist. But I'm not too familiar with the channel workflow...